implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+        );
+
+        return DimensionList.fromCollection(concatenated.map(
+                (kv) -> Dimension.create(kv.getKey(), kv.getValue())
+        ).collect(Collectors.toList()));
+    }
+
+    @Override
+    public void export(MeterRegistry registry) {
+        // split the lines by whether or not they exceed the line length limit.
+        Map<Boolean, List<String>> metricLines =
+                registry.getMeters().stream()
+                        .flatMap(this::toMetricLines)
+                        .collect(Collectors.partitioningBy(DynatraceExporterV2::lineLengthBelowLimit));
+
+        // both keys will be present, even if empty.
+        if (!metricLines.get(false).isEmpty()) {
+            logger.info("dropping {} lines that exceed the line length limit of {}", metricLines.get(false).size(), METRIC_LINE_MAX_LENGTH);
+        }
+
+        sendInBatches(metricLines.get(true));
+    }
+
+    private Stream<String> toMetricLines(Meter meter) {
+        return meter.match(
+                this::toGaugeLine,
+                this::toCounterLine,
+                this::toTimerLine,
+                this::toDistributionSummaryLine,
+                this::toLongTaskTimerLine,
+                this::toTimeGaugeLine,
+                this::toFunctionCounterLine,
+                this::toFunctionTimerLine,
+                this::toMeterLine
+        );
+    }
+
+    Stream<String> toGaugeLine(Gauge meter) {
+        return toGauge(meter);
+    }
+
+    Stream<String> toCounterLine(Counter meter) {
+        return toCounter(meter);
+    }
+
+    double minFromHistogramSnapshot(HistogramSnapshot histogramSnapshot) {
+        ValueAtPercentile[] valueAtPercentiles = histogramSnapshot.percentileValues();
+        double min = Double.NaN;
+
+        for (ValueAtPercentile valueAtPercentile : valueAtPercentiles) {
+            if (valueAtPercentile.percentile() == 0.0) {
+                min = valueAtPercentile.value();
+            }
+        }
+        if (Double.isNaN(min)) {
+            logger.warn("0% quantile disabled, could not determine minimum value.");
+        }
+        return min;
+    }
+
+    Stream<String> toTimerLine(Timer meter) {
+        HistogramSnapshot histogramSnapshot = meter.takeSnapshot();
+        double total = histogramSnapshot.total(getBaseTimeUnit());
+        double max = histogramSnapshot.max(getBaseTimeUnit());
+
+        return toSummaryLine(meter, histogramSnapshot, total, max, true);
+    }
+
+    private Stream<String> toSummaryLine(Meter meter, HistogramSnapshot histogramSnapshot, double total, double max, boolean isTimer) {
+        long count = histogramSnapshot.count();
+
+        double min;
+        if (count == 1) {
+            min = max;
+        } else {
+            min = minFromHistogramSnapshot(histogramSnapshot);
+            if (isTimer) {
+                // it seems like percentile values for timers are recorded as nanoseconds.
+                min = min / 1_000_000;
```

Comment:
Can you use `getBaseTimeUnit()` here or for `value(...)` in `minFromHistogramSnapshot`?